cmd/load.go

```diff
@@ -0,0 +1,308 @@
+			rDir = filepath.Dir(rDir)
+		}
+		rDir, err := filepath.Abs(rDir)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error":     err,
+				"directory": rDir,
+			}).Fatal("error getting absolute Path for R directory")
+		}
+
+		load(cfg.Packages, rs, rDir, cfg.Threads, all, json)
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(loadCmd)
+	loadCmd.Flags().Bool("all", false, "load user packages as well as their dependencies")
+	viper.BindPFlag("all", loadCmd.LocalFlags().Lookup("all")) //There doesn't seem to be a way to bind local flags.
+	loadCmd.Flags().Bool("json", false, "output a JSON object of package info at the end")
+	viper.BindPFlag("json", loadCmd.LocalFlags().Lookup("json")) //There doesn't seem to be a way to bind local flags.
+	// loadCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+// Attempts to load all of the packages in userPackages, and (optionally) the dependencies of those packages, in an R
+// session launched from rDir.
+func load(userPackages []string, rs rcmd.RSettings, rDir string, threads int, all, toJson bool) {
+	if toJson {
+		logger.SetLogLevel("fatal") // disable logging
+	}
+
+	log.WithFields(log.Fields{
+		"all_arg" :  all,
+		"json_arg" : toJson,
+	}).Info("attempting to load packages")
+
+	log.Info("getting relevant packages via `pkgr plan`..................")
+
+	rVersion := rcmd.GetRVersion(&rs)
+	_, installPlan, _ := planInstall(rVersion, false)
+
+	log.Info("finished getting packages from `pkgr plan`__________________")
+
+	var toLoad []string
+	if all {
+		toLoad = installPlan.GetAllPackages()
+	} else {
+		toLoad = userPackages
+	}
+
+	report := InitLoadReport(getRSessionMetadata(rs, rDir))
+
+	resultsChannel := make(chan LoadResult, len(toLoad))
+	sem := make(chan int, threads * 2)
+
+	log.WithFields(
+		log.Fields{
+		"num_to_load": len(toLoad),
+		"threads": threads * 2,
+		"all_arg" :  all,
+		"json_arg" : toJson,
+	}).Info("attempting to load packages")
+
+	// Kick off every load request as a goroutine, each of which will wait on the availability of a semaphore wait group.
+	for _, pkg := range toLoad {
+		go attemptLoadConcurrent(
+			LoadRequest {
+			rs,
+			rDir,
+			pkg,
+			},
+			sem,
+			resultsChannel,
+		)
+	}
+
+	for i := 0; i < len(toLoad); i++ {
+		lr := <-resultsChannel
+		log.WithFields(
+			log.Fields{
+				"pkg" : lr.Package,
+				"succeeded": lr.Success,
+			}).Trace("completed load attempt and adding to load report.")
+		report.AddResult(lr.Package, lr)
+	}
+	close(resultsChannel)
+	close(sem)
+
+	if report.Failures == 0 {
+		log.WithFields(log.Fields{
+			"working_directory":       rDir,
+			"user_packages_attempted": "true",
+			"dependencies_attempted":  all,
+		}).Info("all packages loaded successfully")
+	} else {
+		log.WithFields(log.Fields{
+			"working_directory": rDir,
+			"failures": report.Failures,
+		}).Error("some packages failed to load.")
+	}
+
+	if toJson {
+		//getRSessionMetadata(rs, rDir)
+		log.Info("printing load report as JSON")
```

Comment:
one point of printing out json, is making. it parseable by json readers. By logging to info, you're going to corrupt the json of stdout

Reply:
actually this will just be. ano-op because you set log level to fatal at top so can remove this message